doc: Add Azure and DO postgres (#11234)
MaterializeInc/materialize: doc/user/content/guides/postgres-cloud.md

1. The Materialize replica will need access to connect to the upstream database. This is usually controlled by IP address. If you are hosting your own installation of Materialize, add the replica's IP address in the security group for the RDS instance.

>     If you are using Materialize Cloud, you can follow the steps here to get your [Materialize instance's static IP address](/cloud/security/#static-ip-addresses).

Reviewer: ```suggestion
    If you are using Materialize Cloud, you can follow [these steps](/cloud/security/#static-ip-addresses) to get the static IP address of your instance.
```

1. The Materialize replica will need access to connect to the upstream database. This is usually controlled by IP address. If you are hosting your own installation of Materialize, add the replica's IP address in the security group for the DB instance.

>     If you are using Materialize Cloud, you can follow the steps here to get your [Materialize instance's static IP address](/cloud/security/#static-ip-addresses).

Reviewer: ```suggestion
    If you are using Materialize Cloud, you can follow [these steps](/cloud/security/#static-ip-addresses) to get the static IP address of your instance.
```

1. In the Google Cloud Console, set the `cloudsql.logical_decoding` to `on`. This enables logical replication.

> 1. The Materialize replica will need access to connect to the upstream database. In the Google Cloud Console, enable access e upstream database for the [Materialize instance's static IP address](/cloud/security/#static-ip-addresses)

Reviewer: ```suggestion
1. The Materialize replica will need access to connect to the upstream database. In the Google Cloud Console, enable access on the upstream database for the Materialize replica's IP address.

      If you are using Materialize Cloud, you can follow [these steps](/cloud/security/#static-ip-addresses) to get the static IP address of your instance.
```

Reviewer: Not sure if you omitted the initial sentence for a specific reason, but this should be consistent with other instructions.

Author: I thought we were dropping "replica" and using "instance" instead?

Reviewer: Yeah, I just didn't since I made this suggestion before noticing Andy's comment. Edited.

For more information, see the [Cloud SQL](https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication#configuring-your-postgresql-instance) documentation.

## Azure Database for PostgreSQL

> Before you start, note that a database restart will be required after making changes to the database.

Reviewer: In the intro:

> Note that enabling logical replication may affect performance.

Since a restart to apply replication configuration is required in all cases, could it be better to add this to the intro instead of specifying it just under the instructions for Azure?

> Before you start, note that a database restart will be required after making changes to the database.

Reviewer: For other services, this is added as a step, so maybe also worth adding it in here:

> 1. Restart the database so all changes can take effect.

> Before you start, note that a database restart will be required after making changes to the database.

Author: The thing is that the restart happens automatically as soon as you make the change:
<img width="999" alt="image" src="https://user-images.githubusercontent.com/21223421/159279010-b49794d6-c347-4958-b02f-d2936a3f93e8.png">

No separate action is required, so you don't have to click a restart button or anything like that.

This message is just notice so that people are aware that their database will be restarted once they make those changes and can plan for a maintenance window if needed.

> Before you start, note that a database restart will be required after making changes to the database.

Author: Same goes for the rest, as soon as you try to make such a config change you are prompted with a save and restart button
<img width="1510" alt="image" src="https://user-images.githubusercontent.com/21223421/159293948-f63ae788-3475-4d88-b436-f66c3acd89f2.png">

> 1. The Materialize replica will need access to connect to the upstream database. In your Azure portal, go to the Azure Database for PostgreSQL instance and under the "Connections security" section add your [Materialize instance's IP address](/cloud/security/#static-ip-addresses) to the allowed IP addresses list and click on the "Save" button.

Reviewer 2: ```suggestion
1. The Materialize instance will need access to connect to the database. In your Azure portal, go to the Azure Database for PostgreSQL instance and under the "Connections security" section add your [Materialize instance's IP address](/cloud/security/#static-ip-addresses) to the allowed IP addresses list and click on the "Save" button.
```

Reviewer: Same comment here as above, but here you're also defaulting to Materialize Cloud:

> Not sure if you omitted the initial sentence for a specific reason, but this should be consistent with other instructions.

Reviewer: And +1 to Andy's suggestion to drop "replica" and use "instance" instead. This will need to be adapted for the whole page.

> 1. Enable Logical Replication by going to the Azure Database for PostgreSQL instance, then under the "Replication" section click the "Enable Logical Replication" toggle button and click on "Save".

Reviewer: This is specific to setting the configuration through the Azure portal. The instructions should point to the parameter that needs to be toggled and its value, independent of [how](https://docs.microsoft.com/en-us/azure/postgresql/concepts-logical#set-up-your-server) you choose to set it.

1. Create a [publication](https://www.postgresql.org/docs/current/logical-replication-publication.html) with the tables you want to replicate:

    ```sql

>     CREATE PUBLICATION mz_source FOR TABLE table1, table2;

Reviewer: Is it not possible to `CREATE PUBLICATION mz_source FOR ALL TABLES;` due to user constraints? I see this is being used for e.g. Cloud SQL, where you also don't get superuser rights; maybe worth double checking and correcting for other services.

Author: Yep, it is mentioned a couple of lines bellow.
Will try to replicate this with Cloud SQL, it seems to take ages to spin up an instance though
<img width="439" alt="image" src="https://user-images.githubusercontent.com/21223421/159292242-57f7d40e-e3ea-4124-ac24-bc048d71cb18.png">

Author: Ok so this does not affect Cloud SQL, as the users created with built-in authentication are granted the `cloudsqlsuperuser` role, and have the same set of attributes as the postgres user which is not the case with the `doadmin` user.

Reviewer: Thanks for double-checking!

    ```

>      The `mz_source` publication will contain the set of change events generated from the specified tables, and will later be used to ingest the replication stream.

Reviewer: ```suggestion
     The `mz_source` publication will contain the set of change events generated from the specified tables, and will later be used to ingest the replication stream.
     
For more information, see the [Azure Database for PostgreSQL](https://docs.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-logical#pre-requisites-for-logical-replication-and-logical-decoding) documentation.  
```

> ## DigitalOcean Managed Postgres

Reviewer: This section doesn't follow the structure of the other ones at all. Please use the same step-by-step format!

Author: Well, @ruf-io just asked me to get rid of those...

Reviewer: Whatever we land on, it should just be consistent across the whole page.

Author: @ruf-io what is your take on this?

> The Materialize replica will need firewall access to connect to the upstream database. In the DigitalOcean console, add your [Materialize instance's IP address](/cloud/security/#static-ip-addresses) to the [Trusted Source list](https://docs.digitalocean.com/products/databases/postgresql/how-to/secure/#firewalls) for your Managed PostgreSQL Cluster.

Reviewer: Same comment here iwrt assuming a Materialize Cloud instance.

Connect to your PostgreSQL cluster as the `doadmin` user and create a [publication](https://www.postgresql.org/docs/current/logical-replication-publication.html) with the tables you want to replicate:

```sql
CREATE PUBLICATION mz_source FOR TABLE table1, table2;
```

> **Note:** Because the `doadmin` user, is not a superuser, you will not be able to create a publication for **all** tables.

Reviewer: ```suggestion
**Note:** Because the `doadmin` user is not a superuser, you will not be able to create a publication for _all_ tables.
```

The `mz_source` publication will contain the set of change events generated from the specified tables, and will later be used to ingest the replication stream.

If a table that you want to replicate has a primary key defined, you can use your default replica identity value. If a table you want to replicate has no primary key defined, you must set the replica identity value to FULL:

```sql
ALTER TABLE table1 REPLICA IDENTITY FULL;

> ```

Reviewer: I'm afraid this is also universal, and the explanation is way too involved to have in this side guide. Users should reference to the main CDC guide for the non-cloud stuff.

For more information, see the [Cloud SQL](https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication#configuring-your-postgresql-instance) documentation.

## Azure Database for PostgreSQL

Before you start, note that a database restart will be required after making the changes below.

1. The Materialize instance will need access to connect to the upstream database. This is usually controlled by IP address. In your Azure portal, go to the Azure Database for PostgreSQL instance and under the "Connections security" section add your Materialize instance's IP address to the allowed IP addresses list and click on the "Save" button.

    If you are using Materialize Cloud, you can follow [these steps](/cloud/security/#static-ip-addresses) to get the static IP address of your instance.

> 1. In the Azure portal or using the Azure CLI [enable logical replication](https://docs.microsoft.com/en-us/azure/postgresql/concepts-logical#set-up-your-server) for the PostgreSQL instance.

Reviewer: ```suggestion
1. In the Azure portal or using the Azure CLI, [enable logical replication](https://docs.microsoft.com/en-us/azure/postgresql/concepts-logical#set-up-your-server) for the PostgreSQL instance.
```
